### ISSUE TITLE
Support multi-line regex in StringComparison

### ISSUE THREAD

**Issue by bryceschober** [Simplistix/testfixtures#152]
It sure would be nice to support multi-line regex in `StringComparison` objects.
```python
# Maybe something like:
compare(S(".*bar", multiline=True)), "foo\nbar")
# or even:
compare(S("bar", multiline=True)), "foo\nbar")
```

**Comment by cjw296** [comment]
I don't think multi-line will actually give you what you want:

```
>>> re.match(".*bar", "foo\nbar", re.MULTILINE)
>>>
```

I suspect you want DOTALL:

```
>>> re.match(".*bar", "foo\nbar", re.DOTALL)
<re.Match object; span=(0, 7), match='foo\nbar'>
```

...and you can already do that:

```
>>> compare(S("(?s:.*bar)"), "foo\nbar")
>>>
```

It would be a nice-to-have for `StringComparison` to accept a `flags` parameter, and PRs for that would be welcome!

**Comment by cjw296** [comment]
It's Christmas, and this was fun to implement... released in 6.17.0: https://testfixtures.readthedocs.io/en/latest/comparing.html#string-comparison-objects
